rockchip: reliably distribute net interrupts

k-ronny opened the pull request:
On the NanoPI R4S it takes an average of 3..5 seconds for the network devices
to appear in '/proc/interrupts'.
Wait up to 10 seconds to ensure that the distribution of the interrupts
really happens.

Signed-off-by: Ronny Kotzschmar <ro.ok@me.com>

Ansuel commented on target/linux/rockchip/armv8/base-files/etc/hotplug.d/net/40-net-smp-affinity:
```diff
@@ -4,8 +4,14 @@
+	for seconds in 1 2 3 4 5 6 7 8 9 10; do
+		line=$(grep -m 1 "${device}\$" /proc/interrupts) && break
+		sleep 1
+	done
```
can you change this to a pool logic and sleep for 2 second ?

a pool logic = a while loop... the counter increase and the while terminate when a value is reached.

k-ronny replied:
Thanks for the comment, but can you explain to me the reason for this change?

Ansuel replied:
better readable code, just that. Also sleeping for 2 second IMHO should make this function sleep less in most of the case.

Ansuel commented:
@k-ronny squash the 2 commit and this is ready :D thx for the change

k-ronny committed:
rockchip: reliably distribute net interrupts

On the NanoPI R4S it takes an average of 3..5 seconds for the network devices
to appear in '/proc/interrupts'.
Wait up to 10 seconds to ensure that the distribution of the interrupts
really happens.

Signed-off-by: Ronny Kotzschmar <ro.ok@me.com>

Ansuel commented:
Thanks, merged in master with 9b00e9795660f53caf1f4f5fd932bbbebd2eeeb1

0xFelix commented:
Should this also be included in 22.03?

0xFelix commented:
I created a PR https://github.com/openwrt/openwrt/pull/10273
